src/tree/frag.rs

```diff
@@ -26,6 +26,13 @@ impl Frag {
             _ => None,
         }
     }
+
+    pub fn into_base(self) -> Option<(tree::Node, bool)> {
+        match self {
+            Frag::Base(base, root) => Some((base, root)),
+            _ => None,
+        }
+    }
```

Comment:
this lets us avoid cloning the frag to pull out the base, if we're able to have ownership over the frag